mitodl/open-discussions
# CRUD for userlists in TS

#### Pre-Flight checklist


#### What are the relevant tickets?
#3819 

#### What's this PR do?
This PR adds userlist CRUD functionality in `/infinite`, similar to what can currently be done at https://discussions-rc.odl.mit.edu/learn/lists/, but in the new typescript codebase. Cannot add items to lists yet.

#### How should this be manually tested?
Assuming Open Discussions is set up with at least some course data, and you have logged in (for now via  http://od.odl.local:8063)

1. Visit http://od.odl.local:8063/infinite/lists. Try things like:
    - creating new lists *Add a few learning lists and learning paths. The list type should be reflected on the card*
    - submitting an invalid form for list creation (e.g., missing title)
    - editing lists after you've created them
    - deleting a list.

2. Once you've created a few lists, add some courses to them. The new TS codebase does not support this yet. Do it via http://od.odl.local:8063/learn (click the bookmark, then select a list to add to). Also add some things to the special "favorites" list.
3. View http://od.odl.local:8063/infinite/lists again. Your lists should show the course count.
4. At http://od.odl.local:8063/infinite/lists, click a list title to go to the details page for that list. At the list details page (something like `http://od.odl.local:8063/infinite/lists/<id here>`, you should be able to:
    - click individual resources to open a details drawer
    - edit the list data (e.g., title)
    - ⚠️ re-ordering learning paths not yet supported. 

#### Where should the reviewer start?
In terms of code review, I would start with `UserListsListingPage` and the other files in `frontends/infinite-corridor/src/pages/user-lists`.

#### Screenshots (if appropriate)

<img width="400" alt="listing_desktop" src="https://user-images.githubusercontent.com/9010790/218625376-38497a97-d412-47a5-ba04-ce47458f9bb6.png"> <img width="150" alt="listing_mobile" src="https://user-images.githubusercontent.com/9010790/218625380-c92e4b40-c8c5-4810-a531-c3d82aa84b8e.png">

<img width="400" alt="form_desktop" src="https://user-images.githubusercontent.com/9010790/218625460-78070cea-268e-4fdb-ab59-66d50ef35c3b.png"><img width="150" alt="form_mobile" src="https://user-images.githubusercontent.com/9010790/218625464-e24b25a9-9f3f-44cb-8995-9174ffd4a6b9.png">

<img width="400" alt="details_desktop" src="https://user-images.githubusercontent.com/9010790/218625476-e60ed535-79fa-4da2-b2ee-9f28ccb9fe71.png"><img width="150" alt="details_mobile" src="https://user-images.githubusercontent.com/9010790/218625480-3e4344df-ecf7-430e-a980-99f514309839.png">
